packages/rpc/src/client.ts

```diff
@@ -27,46 +31,86 @@ import type {
 	SetUserVoiceSettingsData,
 	RelationShip,
 } from './typings/structs';
-import { RPCCommands, RPCEvents, type LobbyType, RelationshipType } from './typings/types';
-import { pid as getPid, uuid } from './util';
-
-function subKey(event: string, args?: unknown[]) {
-	return `${event}${JSON.stringify(args)}`;
-}
+import {
+	RPCCommands,
+	RPCEvents,
+	type LobbyType,
+	RelationshipType,
+	RPCErrorCodes,
+	PromptType,
+} from './typings/types.js';
+import { pid as getPid } from './util.js';
 
 type Class<T extends new (...args: any[]) => unknown> = T extends new (...args: any[]) => infer R ? R : never;
 
+const MAX_TIME_RANGE = 2_147_483_647_000;
+
 export interface RPCClientOptions {
-	transport?: keyof typeof transports;
+	/**
+	 * Origin link specified in the [Portal](https://discord.com/developers/applications)
+	 */
 	origin?: string;
+	/**
+	 * rpc transport
+	 */
+	transport: keyof typeof transports;
 }
 
 export interface RPCLoginOptions {
+	/**
+	 * Access token
+	 */
+	accessToken: string;
+	/**
+	 * Client ID
+	 */
 	clientId: string;
+	/**
+	 * Client secret
+	 */
 	clientSecret: string;
-	accessToken: string;
-	rpcToken: string;
-	tokenEndpoint: string;
+	prompt: PromptType;
+	/**
+	 * Scopes to authorize with
+	 */
 	scopes: OAuth2Scopes[];
+	/**
+	 * Token endpoint
+	 */
+	tokenEndpoint: string;
+	/**
+	 * Should we use RPC token or npt
+	 */
+	useRPCToken: boolean;
+	/**
+	 * Username to use if user doesn't have a discord account
+	 */
 	username: string;
 }
 
-/**
- * @typedef {RPCClientOptions}
- * @extends {ClientOptions}
- * @prop {string} transport RPC transport. one of `ipc` or `websocket`
- */
-
 /**
  * The main hub for interacting with Discord RPC
- * @extends {EventEmitter}
+ *
+ * @beta
  */
 export class RPCClient extends EventEmitter {
-	public readonly options: RPCClientOptions;
-	private accessToken: string | null;
-	public clientId: string | null;
-	public application: RPCOAuthApplication | null;
-	public user: Partial<APIUser> | null;
+	private accessToken: string | null = null;
+
+	/**
+	 * ID of the application used in this client
+	 */
+	public clientId: string | null = null;
+
+	/**
+	 * Application used in this client
+	 */
+	public application: RPCOAuthApplication | null = null;
+
+	/**
+	 * User used in this application
```

Comment:
maybe rephrase this to `User using the rpc` or something else, the user isn't used in the application but uses the application